### DISCUSSION TITLE
Is there a way to make a pwa using the new app directory?

### DISCUSSION THREAD

**Discussion by nicholashaemmerle1994** [vercel/next.js#46472] [Help]
Hello everyone :)
I'm a new developer and i want to make a pwa as my final project. We already use the new app directory in our final project and i couldnt find anyone or anything on using this kind of directory for a pwa. 
Maybe some of you have some experiences with that.
Looking forward to some feedbacl.
Thank you so much in advance

**Comment by icyJoseph** [comment]
Haven't used it myself, but here you go:

- https://github.com/shadowwalker/next-pwa/issues/424#issuecomment-1332258575

A word of caution with PWA's, they'll make your app behave in unexpected ways. If you are using GetServerSideProps, you'll not see up to date data, on SPA navigation. That's because PWA's by default use stale-while-revalidate, instead you should use network first falling back to cache. https://developer.chrome.com/docs/workbox/modules/workbox-strategies/#network-first-network-falling-back-to-cache

**Reply by nicholashaemmerle1994** [reply]
Thank you so much. This helps me a lot.
